### ISSUE TITLE
Visualize outputs, inner-state Region

### ISSUE THREAD

**Issue by breznak** [numenta/nupic#2250]
This should be easy and quite handy... 

A region that plots its input sources. This is possible to write today easily but is a repetitive and would be nice to have for many experiments. 

I imagine 2 modes: 
* input is a scalar -> plot as a graph (for numeric values, eg input field (float), or anomalyScore, ...)
* input is an array -> plot as a bitmap (with given dimensions), eg for visualizing state of SP/TM columns

Can be used in user-created Networks, or with OPF if CLAModel is hacked (correct sources are added to the region)

**Comment by breznak** [comment]
@subutai @scottpurdy ?

**Comment by david-ragazzi** [comment]
@breznak Wouldn't be this a role for a debugging tool as cerebro or nupic studio?

**Comment by scottpurdy** [comment]
@breznak - What is the use case? Do you want a plot that is updated as the data is processed? I don't think it makes as much sense if you just want to plot the data after feeding a batch through.

**Comment by breznak** [comment]
@scottpurdy @david-ragazzi purely for convenience. Although plotting streaming data in real-time is a nice idea! I intend it for my colleagues, who use HTM but don't want to hack it so much..just to let them see the results.  (NuStudio could do it, but again..more hustle) 
Other option than a region would be a script in `./scripts/` that does the parsing for `.csv` in model_0/inferences and plots certain data. (which would you think is a better approach -  region/script?)

I don't argue it's some must-have functionality, just that I will write it now, so asking if you want it mainline?

**Comment by breznak** [comment]
I found this cool guy's video: 
https://www.youtube.com/watch?v=tw1dmxcI8pU
and code 
https://github.com/rhyolight/online-anomaly-detection

Just want to ask @rhyolight whether the code is the most up-to-date version available?

**Comment by breznak** [comment]
> https://github.com/rhyolight/online-anomaly-detection 

Matt, is it still running for you? I've installed the dependencies (matplotlib, PySide), forced the plot=True and I see it crunching (`Read 2800 lines...`) but no visual graph is shown..

**Comment by rhyolight** [comment]
@breznak Yes, it still works for me (I can see it plotting), and yes that is the latest code.

**Comment by rhyolight** [comment]
@breznak It's `python run.py --plot`

**Comment by rhyolight** [comment]
@breznak Also, sometimes the Python GUI window pops under other windows, so look around in your OS for it...

**Comment by breznak** [comment]
Thanks Matt,
yes, I run `python run.py --plot`, can't see the window anywhere, even
hidden :( Could you try on (K)Ubuntu if you have some spare around?
I've searched for the problem, found this snip [1] (just mentioning
"ion()") and it seems it works for me only with:
-pylab
-show()
-no ion()

Can anyone try the snip on linux and confirm?
Thanks a bunch!

====================
[1] plotting test-case for Ubuntu/linux

import numpy as np
import matplotlib.pyplot as plt
import pylab

# Come up with x and y
x = np.arange(0, 5, 0.1)
y = np.sin(x)

# Just print x and y for fun
print x
print y

# plot the x and y and you are supposed to see a sine curve
#! pylab.ion()  #breaking-line
plt.plot(x, y)

# without the line below, the figure won't show
pylab.show() # FIXME show() works, ion() not!
#FIXME 2: pylab.show() works, pls.show() does not!
#FIXME 3: it seems ion() "breaks" the window , with the #breaking-line
uncommented nothing is shown.
================

**Comment by rhyolight** [comment]
@breznak No time to test in Kubuntu anytime soon, but you might need a [different backend](http://matplotlib.org/faq/usage_faq.html#what-is-a-backend)?
